packages/sdk-react/src/components/products/ProductDetails/ProductEditForm/ProductEditForm.tsx

```diff
@@ -158,7 +140,11 @@ const ProductEditFormBase = (props: IProductEditFormProps) => {
       description: values.description,
     };
 
-    return updateProduct(payload);
+    return productUpdateMutation.mutate(payload, {
+      onSuccess: (product) => {
+        props.onUpdated?.(product);
```

Comment:
in the previous version of the onSuccess callback there was `props.onCanceled();`.
It changes the view mode of the details dialog, from edit to read.
In your current implementation, the form persists after changes are successfully saved.

We should either put `onCancel` back, or introduce another, more explicit variable (`viewMode`, `isEdit`, etc)

Reply:
Thanks! I was expecting someone familiar with this to come along and explain this behavior to me. Don't you mind if I add the `todo` and advice for the refactoring in the code?

Reply:
yes, good for me